src/DynamoCore/Interfaces/IWorkspaceModel.cs

```diff
@@ -21,5 +21,11 @@ public interface IWorkspaceModel
         event Action NodesCleared;
         event Action<ConnectorModel> ConnectorAdded;
         event Action<ConnectorModel> ConnectorDeleted;
+
+        /// <summary>
+        /// Implement to record node modification for undo/redo
+        /// </summary>
+        /// <param name="nodeModel"></param>
+        void RecordModelForModification(ModelBase nodeModel);
```

Comment:
I'm confused if this is just for node modifications, shouldn't the argument type be `NodeModel`. Also, can you give the client more information about how this method is to be used?